nixos/modules/system/boot/stage-2.nix

```diff
@@ -20,6 +20,11 @@ let
       pkgs.util-linux
     ] ++ lib.optional useHostResolvConf pkgs.openresolv);
     fsPackagesPath = lib.makeBinPath config.system.fsPackages;
+    inherit (config.boot) extraSystemdUnitPaths;
```

Comment:
Maybe add this to the above `inherit (config.boot) systemdExecutable` instead of having a separate one?